bundles/binding/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/SoulissBinding.java

```diff
@@ -0,0 +1,326 @@
+import org.openhab.binding.souliss.internal.network.typicals.RefreshHEALTYThread;
+import org.openhab.binding.souliss.internal.network.typicals.RefreshSUBSCRIPTIONThread;
+import org.openhab.binding.souliss.internal.network.typicals.SoulissGenericTypical;
+import org.openhab.binding.souliss.internal.network.typicals.SoulissT11;
+import org.openhab.binding.souliss.internal.network.typicals.SoulissT12;
+import org.openhab.binding.souliss.internal.network.typicals.SoulissT16;
+import org.openhab.binding.souliss.internal.network.typicals.SoulissT19;
+import org.openhab.binding.souliss.internal.network.typicals.SoulissT22;
+import org.openhab.binding.souliss.internal.network.typicals.StateTraslator;
+
+import org.openhab.binding.souliss.internal.network.typicals.SoulissNetworkParameter;
+import org.openhab.binding.souliss.internal.network.udp.SendDispatcherThread;
+import org.openhab.binding.souliss.internal.network.udp.UDPServerThread;
+
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.osgi.service.event.Event;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class load parameters from cfg files (method updated)
+ * 
+ * @author Tonino Fazio
+ * @since 1.7.0
+ */
+public class SoulissBinding<E> extends
+		AbstractActiveBinding<SoulissBindingProvider> implements ManagedService {
+
+	private static Logger LOGGER = LoggerFactory
+			.getLogger(SoulissBinding.class);
+
+	/** to keep track of all binding providers */
+
+	protected EventPublisher eventPublisher = null;
+
+	@Override
+	public void setEventPublisher(EventPublisher eventPublisher) {
+		this.eventPublisher = eventPublisher;
+	}
+
+	@Override
+	public void unsetEventPublisher(EventPublisher eventPublisher) {
+		this.eventPublisher = null;
+	}
+
+	@Override
+	public void bindingChanged(BindingProvider provider, String itemName) {
+		// TODO Auto-generated method stub
+		super.bindingChanged(provider, itemName);
+	}
+
+	@Override
+	public void activate() {
+		// TODO Auto-generated method stub
+		super.activate();
+	}
+
+	@Override
+	public void deactivate() {
+		// TODO Auto-generated method stub
+		super.deactivate();
+	}
+
+	/**
+	 * Read parameters from cfg file
+	 * 
+	 * @author Tonino Fazio
+	 * @since 1.7.0
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config)
+			throws ConfigurationException {
+		if (config != null) {
+			Enumeration<String> enumConfig = config.keys();
+
+			while (enumConfig.hasMoreElements()) {
+				String sName = enumConfig.nextElement();
+				LOGGER.info("PARAMETER: " + sName + " = "
+						+ (String) config.get(sName));
+				switch (sName) {
+				case "IP_LAN":
```

Comment:
using switch-case on Strings unnecessarily introduces a dependency on JDK1.7. If there are no good reasons we'd try to stick to java 1.5